direct/io-bulk-fs/src/main/java/cz/o2/proxima/direct/bulk/Reader.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.bulk;
+
+import cz.o2.proxima.annotations.Internal;
+import cz.o2.proxima.storage.StreamElement;
+
+/** Proxima's representation of generic reader. */
+@Internal
+public interface Reader extends AutoCloseable, Iterable<StreamElement> {
```

Comment:
Is this supposed to be reiterable? If not can we change this to `Iterator` or `Stream`?

Reply:
I think there should be no problem with reiterating this. It is not possible to reiterate it twice at the same time, but there is no reason for that. I'd keep it as it is, because it is @Internal API and it is handy to have it `Iterable`.